log/logsink.go

```diff
@@ -108,8 +108,8 @@ var TimestampFunc = func() string {
 	return time.Now().UTC().Format(time.RFC3339Nano)
 }
 
-// Logger writes logs to a specified output
-type Logger struct {
+// Sink writes logs to a specified output
+type Sink struct {
```

Comment:
Is the `Sink` something users of this library should be able to use? If not, should we move it to an internal package or make it private?

Reply:
Not really. This is from the `logr` docs:

> LogSink represents a logging implementation. End-users will generally not interact with this type. 

So a `Sink` can either be interpreted as an internally hidden or advanced settings.

Reply:
In that case I think I would try to reduce the public API of this library again and move the `Sink` implementation to an `internal/sink` package. We can still reintroduce it in a 2.y if we see the need for it.